### ISSUE TITLE
Add support to build packages for non-LTS kernel versions

### ISSUE THREAD

**Issue by markusressel** [jbrodriguez/arch-zfs-docker#3]
Hi there,

thanks for doing what I was about to to myself :partying_face: :heart: 

Since I am not using the LTS kernels, it would be awesome if it was possible to have a Parameter to switch between LTS and non-LTS kernel targets.

Do you think this is feasible?

**Comment by jbrodriguez** [comment]
hi, it think it's feasible, not an expert in making a Dockerfile configurable though, so can't help that much.

probably the best bet at the moment, would be to fork and remove -lts as applicable, then test, it should kind of work

**Comment by markusressel** [comment]
Wow ok, so I have no Idea how you were able to successfully build both the zfs-util as well as the zfs-linux and zfs-linux-headers packages using the files in this repository :rofl: After trying to adapt the files for the non-LTS kernel version and running into one issue after the next I have essentially touched everything in every file.... but I have successfully built all of the above mentioned packages:

```bash
├── zfs-linux
│   ├── zfs-linux-2.2.6_6.10.10.arch1.1-1-x86_64.pkg.tar
│   └── zfs-linux-headers-2.2.6_6.10.10.arch1.1-1-x86_64.pkg.tar
└── zfs-utils
    ├── zfs-utils-2.2.6-1-x86_64.pkg.tar
    └── zfs-utils-debug-2.2.6-1-x86_64.pkg.tar
```

I will try to install these files now and hopefully not brick my system.
In case I do, the recovery will probably take a while :sweat_smile: 

About forking this repo:
I am not sure if forking is the best way going forward, since I essentially have changed at least 50% of the code/files. I am still unsure myself about what I want to do with this going forward. The "official" repos seem to be maintained pretty slowly these days, which leads to me not being able to update, and I want a way around that. However, I also don't really want to build these packages each and every time myself... I guess creating my own repo would be a solution, but I am not sure if I want to invest in that.

Anyway, thx for giving me the trigger to pull through with this and get it going. Its still not perfect, but successfully build package artefacts is a hell of a lot more than I had hoped for :smile:

**Comment by jbrodriguez** [comment]
> Wow ok, so I have no Idea how you were able to successfully build both the zfs-util as well as the zfs-linux and zfs-linux-headers packages using the files in this repository

well, that was easy, just following the instructions in the README 🤷‍♂️

i went the lts route because it's supported by openzfs right now, newer kernels may not be supported initially or have some bug (which has already happened)

ymmv

**Comment by markusressel** [comment]
I didn't expect such a difference between LTS and non LTS, but the issues I encountered were not only because of that. F.ex. the sed command to update the PKGBUILD file did not work for me. Also there was a commented out line in the script that ommitted the installation of the zfs-utils package, which was needed in my case (at least it helped along the way). Very weird stuff everywhere, feel free to take a look at the end result in my fork if you are interested. Again I am not sure yet how I will proceed with this.

I used the above mentioned packages to install/update my desktop system, and I get a Kernel panic at boot. However, I also had issues with a full disk, so I cannot tell for sure where the issue lies. I have to invest more time into debugging this some other day. Not sure if it is of interest in this issue, but I can post an update when I get it to work.

**Comment by markusressel** [comment]
@jbrodriguez Update: It works!

The issue I had was simply related to a missing `zfs export <pool>` call before rebooting.
Not sure why this was necessary this time, since the machine is the same and "nothing" changes besides the approximately 1000 package updates :sweat_smile: but simply booting a recovery image from [here](https://archzfs.leibelt.de/), importing the pools using `zpool import -f <pool>` and exporting them again right after using `zpool export <pool>` fixed the issue. I can successfully boot the machine now using the latest kernel and zfs versions:

```bash
> uname -a
Linux Silver 6.10.10-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 12 Sep 2024 17:21:02 +0000 x86_64 GNU/Linux
> zfs --version
zfs-2.2.6-1
zfs-kmod-2.2.6-1
> zpool list
NAME    SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
bpool  3.75G   159M  3.59G        -         -     2%     4%  1.00x    ONLINE  -
rpool   920G   729G   191G        -         -    43%    79%  1.00x    ONLINE  -
```

I will reuse the same setup on my laptop and use that to optimize the procedure, maybe this can actually become something thats easy to use.

**Comment by markusressel** [comment]
I have spent some time on my fork and it can now support both LTS as well as "non-LTS" and puts the packages right into a custom repository path. Feel free to take a look: https://github.com/markusressel/arch-zfs-docker

Since I now have what I asked for in this issue, I will close it now.

Thx again for your work! I have added an attribution to the bottom of the README (in addition to "officially" forking your repo).
